flink-core/src/main/java/org/apache/flink/core/failure/FailureEnricher.java

```diff
@@ -84,14 +88,26 @@ enum FailureType {
          * Get the ID of the job.
          *
          * @return the ID of the job
+         * @deprecated This method is deprecated since Flink 1.19. All metadata about the job should
+         *     be provided uniformly by {@link #getJobInfo()}.
+         * @see <a
+         *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-382%3A+Unify+the+Provision+of+Diverse+Metadata+for+Context-like+APIs">
+         *     FLIP-382: Unify the Provision of Diverse Metadata for Context-like APIs </a>
          */
+        @Deprecated
```

Comment:
Same here for the deprecated methods.

Reply:
For these deprecated methods in the modified interfaces, I've changed them from abstract to default and provide default implementations.